compiler/optimizer/VPConstraint.cpp

```diff
@@ -1353,28 +1353,15 @@ TR::VPKnownObject *TR::VPKnownObject::create(OMR::ValuePropagation *vp, TR::Know
    //
    constraint = NULL;
 #ifdef J9_PROJECT_SPECIFIC
-      {
-      TR::VMAccessCriticalSection vpKnownObjectCriticalSection(vp->comp(),
-                                                                TR::VMAccessCriticalSection::tryToAcquireVMAccess);
 
-      if (vpKnownObjectCriticalSection.hasVMAccess())
-         {
-         TR_OpaqueClassBlock *clazz = TR::Compiler->cls.objectClass(vp->comp(), knot->getPointer(index));
-         TR_OpaqueClassBlock *jlClass = vp->fe()->getClassClassPointer(clazz);
-         if (isJavaLangClass)
-            {
-            TR_ASSERT(clazz == jlClass, "Use createForJavaLangClass only for instances of java/lang/Class");
-            clazz = TR::Compiler->cls.classFromJavaLangClass(vp->comp(), knot->getPointer(index));
-            }
-         else
-            {
-            // clazz is already right
-            TR_ASSERT(clazz != jlClass, "For java/lang/Class instances, caller needs to use createForJavaLangClass.");
-            }
-         constraint = new (vp->trStackMemory()) TR::VPKnownObject(clazz, vp->comp(), index, isJavaLangClass);
-         vp->addConstraint(constraint, hash);
-         }
-      }
+   TR_OpaqueClassBlock *clazz = vp->comp()->fej9()->getObjectClassFromKnownObjectIndex
+      (vp->comp(), index, isJavaLangClass);
```

Comment:
This implementation eliminates the assert that made sure that, whenever the parameter `isJavaLangClass` is true, clazz is the class the java/lang/class